homeassistant/components/alarm_control_panel/nx584.py

```diff
@@ -31,13 +49,12 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 class NX584Alarm(alarm.AlarmControlPanel):
     """Represents the NX584-based alarm panel."""
 
-    def __init__(self, hass, host, name):
+    def __init__(self, hass, url, name):
         """Initalize the nx584 alarm panel."""
         from nx584 import client
         self._hass = hass
-        self._host = host
         self._name = name
-        self._alarm = client.Client('http://%s' % host)
+        self._alarm = client.Client(url)
```

Comment:
```python
self._url = url
self._alarm = client.Client(self._url)
```
You're not storing `url` in `self` - that's why linting is complaining about line 86